music21/musicxml/m21ToXml.py

```diff
@@ -3277,7 +3278,9 @@ def parseFlatElements(
                         self.parseOneElement(obj, AppendSpanners.NORMAL)
 
             for n in notesForLater:
-                if n.isRest and n.style.hideObjectOnPrint and n.duration.type == 'inexpressible':
+                if (n.isRest
+                        and n.style.hideObjectOnPrint
+                        and n.duration.type in ('inexpressible', 'complex')):
```

Comment:
Greg -- 'complex' durations don't exist in m21ToXml unless makeNotation=False.  I don't want fixes in here that cause different code paths for the two -- a complex duration in a stream at this point is an error.